apps/req.c

```diff
@@ -803,9 +818,12 @@ int req_main(int argc, char **argv)
             if (!X509_set_issuer_name(new_x509, issuer))
                 goto end;
             if (days == UNSET_DAYS) {
-                days = DEFAULT_DAYS;
+                if (not_after == NULL)
+                    days = DEFAULT_DAYS;
+                else
+                    days = 0;
             }
```

Comment:
This should be sufficient:
```
if (days == UNSET_DAYS)
    days = DEFAULT_DAYS;
```
because `days` is irrelevant if `not_after != NULL`.
Moreover, you can integrate this with the above conditional warning as follows:
```
        if (days != UNSET_DAYS)
            BIO_printf(bio_err, "Warning: Ignoring -days without -x509; not generating a certificate\n");
        else
            days = DEFAULT_DAYS;
```

Reply:
The first proposal is applied.

I disagree with the second proposal.
IMHO the value checking is best placed in a single location, whereas the proposed location is solely used for checking the dependency with the `gen_x509` option. Furthermore, the original condition was extended with an `else if` branch i the meantime.